File: typescript/docs_code_snippets/knowledge-get-by-name.ts
import { Opper } from "opperai";

const opper = new Opper();

// setup
const kbName = `docs-snippet-${Date.now()}`;
const created = await opper.knowledge.create({ name: kbName });
// /setup

// --- docs ---
const kb = await opper.knowledge.getByName(kbName);
console.log(`Name: ${kb.name}, Docs: ${kb.count}, Model: ${kb.embedding_model}`);
// --- /docs ---

// cleanup
await opper.knowledge.delete(created.id);

File: python/docs_code_snippets/knowledge-get-by-name.py
import time
from opperai import Opper

opper = Opper()

# Setup
kb_name = f"docs-snippet-{int(time.time())}"
kb = opper.knowledge.create(name=kb_name)

# --- docs ---
kb = opper.knowledge.get_by_name(kb_name)
print(f"Name: {kb.name}, Docs: {kb.count}, Model: {kb.embedding_model}")
# --- /docs ---

# Cleanup
opper.knowledge.delete(kb.id)
